cdap-common/src/main/java/org/apache/twill/filesystem/FileContextLocation.java

```diff
@@ -80,6 +80,32 @@ public boolean createNew() throws IOException {
   }
 
   @Override
+  public boolean createNew(String permission) throws IOException {
```

Comment:
We added new methods to the Location interface. I copied the implementation from Twill.